.github/workflows/build.yml

```diff
@@ -18,60 +18,68 @@ jobs:
   build:
     strategy:
       matrix:
-        runtime: [ linux-x64, linux-arm64, linux-arm, win-x64, win-arm64, osx-x64, osx-arm64 ]
+        runtime: [ linux-s390x, linux-ppc64le, linux-x64, linux-arm64 ]
         include:
+
         - runtime: linux-x64
-          os: ubuntu-latest
-          devScript: ./dev.sh
+          runs_on: amd64
 
         - runtime: linux-arm64
-          os: ubuntu-latest
-          devScript: ./dev.sh
+          runs_on: arm64
 
-        - runtime: linux-arm
-          os: ubuntu-latest
-          devScript: ./dev.sh
+        - runtime: linux-ppc64le
+          runs_on: ppc64le
+          patch_tag: ppc64le
 
-        - runtime: osx-x64
-          os: macOS-latest
-          devScript: ./dev.sh
+        - runtime: linux-s390x
+          runs_on: s390x
+          patch_tag: s390x
 
-        - runtime: osx-arm64
-          os: macOS-latest
-          devScript: ./dev.sh
+    runs-on: [ self-hosted, noble, "${{ matrix.runs_on }}" ]
```

Comment:
As discussed, we might need to think about disaster recovery. Since this is relying on there being a working build of the self-hosted runner application.